lib/src/Pos/Launcher/Scenario.hs

```diff
@@ -89,15 +87,10 @@ runNode' NodeResources {..} workers' plugins' = ActionSpec $ \diffusion -> ntpCh
     let unpackPlugin (ActionSpec action) =
             action diffusion `catch` reportHandler
 
-    -- Either all the plugins are cancelled in the case one of them
-    -- throws an error, or otherwise when the shutdown signal comes,
-    -- they are killed automatically.
-    void
-      (race
-           (void (mapConcurrently (unpackPlugin) $ workers' ++ plugins'))
-           waitForShutdown)
+    void (mapConcurrently (unpackPlugin) $ workers' ++ plugins')
+
+    exitFailure
```

Comment:
Why is it `exitFailure`?

Reply:
Currently we expect at least some workers and plugins to work indefinitely, so this `exitFailure` shouldn't be reachable.
If we reach this `exitFailure`, it likely indicates that something went wrong (because it means that all workers and plugins stopped and not because of exceptions).
Maybe it makes more sense to `reportFatalError` (with explanation) in this case (which will eventually lead to `exitFailure` anyway).

Reply:
Got it. Please add some comment to this line, because it's not obvious that we use `exitOnShutdown` outside.

Reply:
@hithroc I am going to merge this PR, but please make sure @volhovm is happy (in another PR to `develop`) 😉